hack/update-codegen.sh

```diff
@@ -66,6 +66,7 @@ applyconfigurationgen_external_apis+=("k8s.io/apimachinery/pkg/apis/meta/v1")
 applyconfigurationgen_external_apis_csv=$(IFS=,; echo "${applyconfigurationgen_external_apis[*]}")
 applyconfigurations_package="k8s.io/client-go/applyconfigurations"
 ${applyconfigurationgen} \
+  --openapi-schema "${KUBE_ROOT}/api/openapi-spec/swagger.json" \
```

Comment:
Thanks for catching this. I spent quite a while reviewing how everything works and I've come up with a solution that seems to avoid falling into any dependency cycle pits:

Instead of depending on `swagger.json`, I'm going to depend on a build artifact that is produced much earlier-- the openAPI `definitions` that are produced by `openapi-gen` and written to `zz_generated.openapi.go`. This artifact includes only the `definitions` (aka models, aka schemas) and not the `endpoints`, but that's all we need.

This avoids introducing a cycle between `update-openapi-spec` and `update-codegen`. Moving `update-openapi-spec` to run before `update-codegen` would have introduced a cycle because `update-openapi-spec` runs the kube-apiserver and curls it to create `swagger.json`, and the kube-apiserver build process depends on the code `update-codegen` produces.